#486 - Refactoring project structure (3.3.0)

astier opened the pull request:
- Use PropertyModel instead of raw data

reckart commented on webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java:
```diff
@@ -275,8 +248,7 @@ public void onSubmit(AjaxRequestTarget aTarget, Form<?> aForm)
 
-        add(layerSelector = new AnnotationFeatureForm.LayerSelector
-            ("defaultAnnotationLayer", annotationLayers));
+        add(layerSelector = new LayerSelector(this, "defaultAnnotationLayer", annotationLayers));
```
Instead of directly passing the `annotationLayers` as a list, an IModel instance should be passed.

astier replied:
Can we make a separate issue for this? I am not familiar with IModel. It also wasn't me who implemented this. Maybe we can fix this an other time.

reckart commented on webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/LayerSelector.java:
```diff
@@ -0,0 +1,106 @@
+            @Override
+            protected void onUpdate(AjaxRequestTarget aTarget)
+            {
+                AnnotationDetailEditorPanel editorPanel = featureForm.getEditorPanel();
```
Instead of passing the AnnotationDetailEditorPanel, passing an AnnotationActionHandler (which is implemented by AnnotationDetailEditorPanel) would seem more appropriate. Also passing an IModel<AnnotatorState> separately instead of fetching it from the ADEP.

astier replied:
It seems like the ADEP is needed in this method, because an ADEP specific method is called at line 103:
```
editorPanel.clearFeatureEditorModels(aTarget);
```
 Also, the ADEP is passed as an argument at line 80 to the DeleteOrReplaceAnnotationModalPanel instance what has to be also of type ADEP because the DeleteOrReplaceAnnotationModalPanel instance calls ADEP specific methods.

reckart replied:
True. I'll see how far I can clean up/push this when I merge.

astier commented on webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/AnnotationFeatureForm.java:
```diff
@@ -282,7 +282,33 @@ protected void onConfigure()
+
+            @Override
+            public void detach()
+            {
```
Not sure what I should do here.

reckart replied:
Use something like PropertyModel.of(this, "annotationLayers")?

astier commented:
I dropped my original changes where I refactored LayerSelector.class in its own file. Instead I tried to follow your request and to make use of an IModel instead of passing the raw data to the DropDownChoice instance. I hope I've done it right.

astier commented:
I changed IModel to PropertyModel like you suggested.

astier commented:
This one is ready.
